files/www/luci-static/resources/view/network/wireless.js

```diff
@@ -0,0 +1,2270 @@
+		uci.unset('wireless', id, 'disabled');
+		uci.unset('wireless', radio, 'disabled');
+	}
+	else {
+		uci.set('wireless', id, 'disabled', '1');
+
+		var all_networks_disabled = true,
+		    wifi_ifaces = uci.sections('wireless', 'wifi-iface');
+
+		for (var i = 0; i < wifi_ifaces.length; i++) {
+			if (wifi_ifaces[i].device == radio && wifi_ifaces[i].disabled != '1') {
+				all_networks_disabled = false;
+				break;
+			}
+		}
+
+		if (all_networks_disabled)
+			uci.set('wireless', radio, 'disabled', '1');
+	}
+
+	return map.save().then(function() {
+		ui.changes.apply()
+	});
+}
+
+function next_free_sid(offset) {
+	var sid = 'wifinet' + offset;
+
+	while (uci.get('wireless', sid))
+		sid = 'wifinet' + (++offset);
+
+	return sid;
+}
+
+function add_dependency_permutations(o, deps) {
+	var res = null;
+
+	for (var key in deps) {
+		if (!deps.hasOwnProperty(key) || !Array.isArray(deps[key]))
+			continue;
+
+		var list = deps[key],
+		    tmp = [];
+
+		for (var j = 0; j < list.length; j++) {
+			for (var k = 0; k < (res ? res.length : 1); k++) {
+				var item = (res ? Object.assign({}, res[k]) : {});
+				item[key] = list[j];
+				tmp.push(item);
+			}
+		}
+
+		res = tmp;
+	}
+
+	for (var i = 0; i < (res ? res.length : 0); i++)
+		o.depends(res[i]);
+}
+
+var CBIWifiFrequencyValue = form.Value.extend({
+	callFrequencyList: rpc.declare({
+		object: 'iwinfo',
+		method: 'freqlist',
+		params: [ 'device' ],
+		expect: { results: [] }
+	}),
+
+	load: function(section_id) {
+		return Promise.all([
+			network.getWifiDevice(section_id),
+			this.callFrequencyList(section_id)
+		]).then(L.bind(function(data) {
+			this.channels = {
+				'2g': L.hasSystemFeature('hostapd', 'acs') ? [ 'auto', 'auto', true ] : [],
+				'5g': L.hasSystemFeature('hostapd', 'acs') ? [ 'auto', 'auto', true ] : [],
+				'6g': L.hasSystemFeature('hostapd', 'acs') ? [ 'auto', 'auto', true ] : [],
+				'60g': []
+			};
+
+			for (var i = 0; i < data[1].length; i++) {
+				if (!data[1][i].band)
+					continue;
+
+				var band = '%dg'.format(data[1][i].band);
+
+				this.channels[band].push(
+					data[1][i].channel,
+					'%d (%d Mhz)'.format(data[1][i].channel, data[1][i].mhz),
+					!data[1][i].restricted
+				);
+			}
+
+			var hwmodelist = L.toArray(data[0] ? data[0].getHWModes() : null)
+				.reduce(function(o, v) { o[v] = true; return o }, {});
+
+			this.modes = [
+				'', 'Legacy', hwmodelist.a || hwmodelist.b || hwmodelist.g,
+				'n', 'N', hwmodelist.n,
+				'ac', 'AC', L.hasSystemFeature('hostapd', '11ac') && hwmodelist.ac,
+				'ax', 'AX', L.hasSystemFeature('hostapd', '11ax') && hwmodelist.ax
+			];
+
+			var htmodelist = L.toArray(data[0] ? data[0].getHTModes() : null)
+				.reduce(function(o, v) { o[v] = true; return o }, {});
+
+			this.htmodes = {
+				'': [ '', '-', true ],
+				'n': [
+					'HT20', '20 MHz', htmodelist.HT20,
+					'HT40', '40 MHz', htmodelist.HT40,
+					'VHT20', '20 MHz', htmodelist.VHT20,
+					'VHT40', '40 MHz', htmodelist.VHT40,

```

Comment:
Perhaps it makes sense to mark these as non-standard (e.g., `'*VHT20'`) and add a remark (similar to the Force 40 MHz remark).
(clearly now this is not important)

Reply:
I'm not sure, why isn't this done in `this.bands`?

Reply:
> Perhaps it makes sense to mark these as non-standard (e.g., `'*VHT20'`) and add a remark (similar to the Force 40 MHz remark). (clearly now this is not important)

Also worth noting in that remark is that VHT40 will not work well (or at all) in 2.4Ghz. So this is something that needs to be noted. VHT40 and VHT80 only work well in 5Ghz because of low non-Wifi interference.

Reply:
One more `，`